lib/wrapeth/provider.ex

```diff
@@ -1,86 +1,72 @@
+        {:ok, accounts} = call_client(:eth_accounts)
         accounts
       end
 
       @impl true
-      def get_balance(address, block \\ "latest") do
-        {module_name, node_url} = get_module_and_url()
-        {:ok, balance} = module_name.eth_get_balance(address, block, url: node_url)
+      def eth_get_balance(address, block \\ "latest", _args \\ []) do
+        {:ok, balance} = call_client(:eth_get_balance, [address, block])
         balance
       end
 
       @impl true
-      def get_block_number() do
-        {module_name, node_url} = get_module_and_url()
-        {:ok, block_number} = module_name.eth_block_number(url: node_url)
+      def eth_block_number(_args \\ []) do
+        {:ok, block_number} = call_client(:eth_block_number)
         block_number
       end
 
       @impl true
-      def get_block_by_number(number, full) do
-        {module_name, node_url} = get_module_and_url()
-        {:ok, block} = module_name.eth_get_block_by_number(number, full, url: node_url)
+      def eth_get_block_by_number(number, full, _args \\ []) do
+        {:ok, block} = call_client(:eth_get_block_by_number, [number, full])
         block
       end
 
       @impl true
-      def get_transaction_count(address, block \\ "latest") do
-        {module_name, node_url} = get_module_and_url()
-        {:ok, tx_count} = module_name.eth_get_transaction_count(address, block, url: node_url)
+      def eth_get_transaction_count(address, block \\ "latest", _args \\ []) do
+        {:ok, tx_count} = call_client(:eth_get_transaction_count, [address, block])
         tx_count
       end
 
       @impl true
-      def get_block_transaction_count_by_number(block \\ "latest") do
-        {module_name, node_url} = get_module_and_url()
-
-        {:ok, tx_count} =
-          module_name.eth_get_block_transaction_count_by_number(block, url: node_url)
+      def eth_get_block_transaction_count_by_number(block \\ "latest", _args \\ []) do
+        {:ok, tx_count} = call_client(:eth_get_block_transaction_count_by_number, [block])
 
         tx_count
       end
 
       @impl true
-      def get_gas_price() do
-        {module_name, node_url} = get_module_and_url()
-
-        {:ok, gas} =
-          module_name.eth_gas_price(url: node_url)
+      def eth_gas_price(_args \\ []) do
+        {:ok, gas} = call_client(:eth_gas_price)
 
         gas
       end
 
       @impl true
-      def get_transaction_receipt(hash) do
-        {module_name, node_url} = get_module_and_url()
-
-        {:ok, receipt} =
-          module_name.eth_get_transaction_receipt(hash, url: node_url)
+      def eth_get_transaction_receipt(hash, _args \\ []) do
+        {:ok, receipt} = call_client(:eth_get_transaction_receipt, [hash])
 
         receipt
       end
+
+      defp call_client(method_name, args \\ [], _arg \\ []) do
+        case apply(@client_type, method_name, args ++ [[{:url, @node_url}]]) do
+          {:ok, value} -> {:ok, value}
+          {:error, error} -> raise(error)
```

Comment:
What do you think it's best here? Raising or returning an error?

Reply:
You are right. it's a better idea to return the error. This way, the future user of the tool can have the flexibility to decide how to handle it.